### DISCUSSION TITLE
Multiple Currencies / Multiple Companies / Multiple Chart of accounts templates support

### DISCUSSION THREAD

**Discussion by bugking** [arrobalytics/django-ledger#346] [Ideas]
Curiously is there any plan for support following features?

1. Multiple currencies:
   * if a company has multiple currencies, and the financial report requires to convert other currencies to the dominate currency.
2. Multiple companies:
   * consoled cash flow & P&L & Balance sheet
3. More than one chart of accounts templates
   * this is useful to adapt to the situation of multiple companies in different regions.

**Comment by elarroba** [comment]
@bugking Thanks for your inquiry.

There is a plan to support all of those features. Please check our [ROADMAP](https://github.com/arrobalytics/django-ledger/blob/develop/ROADMAP.md) document for more details. In particular, your feature request will be addressed in v0.7 and v0.9.

If you don't mind, can you elaborate on bullet 3? The data model allows for one CoA (Chart of Accounts) per Entity Model (company). I have considered multiple CoA's per entity (and it's possible to do), but didn't find any practical situations for that scenario (can always create more accounts if needed). In your case, you describe the scenario of multiple companies in different regions, but they would still be different companies, (each with its own CoA). So, just want to make sure I understand your request and possibly plan for that.

Thanks!
Miguel

**Comment by bugking** [comment]
my third request is about the CoA template per region or country, but one company still has one CoA. sorry if my previous request misled u ;)

**Comment by JoshNourse** [comment]
Great project, been experimenting with Beancount but looking forward to checking this out!  

To chime in on COA's: I can see the use in this as not every account should be made available for every region or department.  For instance, the warehouse does not need an account to hold expenses for design software.  And different countries will have different compliance requirements.  For instance in the U.S. it is important to have the correct payroll accounts to reconcile to the annual 940.

Could this be accomplished with "Divisions" with only a subset of the full COA available for the division. In the end you are going to need consolidated financials so all the pieces need to fit together.

I typically create multiple company /  departments by means of a coded account number.  I have used software in the past where this type of structure was actually enforced when creating account numbers.

**i.e.  Account 6140310 - Office supplies**
6: Signifies an expense account
1: Company or Division
40: Department
310: Office Supplies

The Portugal variant could be 6240310.

Then it comes down to a reporting issue and being able to parse the account number for the structure to generate the correct reports or allow user controls to groups of accounts.
